### ISSUE TITLE
Cannot install from source Python 3.7 macOS - 'extras_require' must be a dictionary whose values are strings or lists of strings containing valid project/version requirement specifiers.

### ISSUE THREAD

**Issue by adamtheturtle** [takluyver/flit#214]
Hello, `flit` is a dependency of a dependency in the application I maintain.

The following is the root cause of issues our users have been seeing.
Using pre-built wheels is not an option in this case.
The URL was taken from https://pypi.org/project/flit/#files.

```
(flit-example) Adam@MacBook-Pro ~/D/example> curl -O https://files.pythonhosted.org/packages/72/9d/b8f00445bb458a15c670308cf30e08900dc749690f299b7539ab70273ac2/flit-1.1.tar.gz
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
100 90706  100 90706    0     0   152k      0 --:--:-- --:--:-- --:--:--  152k
(flit-example) Adam@MacBook-Pro ~/D/example> tar -xzf *.tar.gz
(flit-example) Adam@MacBook-Pro ~/D/example> cd flit-1.1/
(flit-example) Adam@MacBook-Pro ~/D/e/flit-1.1> python
Python 3.7.0 (default, Jun 29 2018, 20:13:13) 
[Clang 9.1.0 (clang-902.0.39.2)] on darwin
Type "help", "copyright", "credits" or "license" for more information.
>>> 
(flit-example) Adam@MacBook-Pro ~/D/e/flit-1.1> pip install .
Processing /Users/Adam/Desktop/example/flit-1.1
  Missing build requirements in pyproject.toml for file:///Users/Adam/Desktop/example/flit-1.1.
  This version of pip does not implement PEP 517 so it cannot build a wheel without 'setuptools' and 'wheel'.
  Installing build dependencies ... done
    Complete output from command python setup.py egg_info:
    error in flit setup command: 'extras_require' must be a dictionary whose values are strings or lists of strings containing valid project/version requirement specifiers.
    
    ----------------------------------------
Command "python setup.py egg_info" failed with error code 1 in /private/var/folders/3l/ffdytd9n2cnbgch4rwt6k7jm0000gn/T/pip-req-build-0kpca0if/
```

My guess is that this is a bug in `flit`.

Thank you,
Adam Dangoor

**Comment by takluyver** [comment]
Yup, it is, sorry. I'll put a fix out soon.

**Comment by adamtheturtle** [comment]
Thank you @takluyver . I'm currently working on thinking of a workaround. If "soon" is going to be really soon then I won't do that. Of course no expectations from you, but do you have any estimate of how long it might take to put out a fix?

**Comment by takluyver** [comment]
It's already committed, but I was waiting on another package going through conda-forge to verify the fix. This sounds like it's easier to hit than I thought.

**Comment by takluyver** [comment]
Verified that I can't `pip install` the flit 1.1 sdist from PyPI, but I can with an sdist built from master. So I just need to update the release notes and make a new release.

**Comment by takluyver** [comment]
Flit 1.2 is out, this should fix the problem.

**Comment by adamtheturtle** [comment]
Thank you @takluyver . That was very fast, I am now testing DC/OS E2E to see if it fixes our issue.
